syntax = "proto3";
package IM.BaseDefine;
option optimize_for = LITE_RUNTIME;

//登录信息
message LoginInfo
{
	 string loginname = 1;  //登录的用户名
	 string password =  2;  //登录的密码
	 sint64 tlogin  = 3;	//登录的时间(此为聊天服务器的时间而不是客户端时间)
	 string clientguid = 4; //设备的guid(这个又各端自己生成)
	 int32  clinettype = 5; //登录的客户端类型 默认为0为未知设备 1为pc端 2为Android 3为iphone 4为Mac
	 string clientOs = 6;   //pc为windows版本 Android为Android版本 IOS为IOS版本
	 string clientDev = 7;  //设备名称(比如小米MIX2 这类的名称)
};

message RegisterInfo
{
	 string cellphone = 1;  //用户注册的手机号
	 string email =  2;  	//用户邮箱
	 string username = 3;	//用户名
	 string password = 4;	//用户的密码
	 sint64 tRegister  = 5; //用户注册的时间(此为聊天服务器的时间而不是客户端时间)
};

//会话类型
enum SessionType
{
	defaultChat = 0;			//初始值无意义
	SingleChat  = 1;          	//单聊会话
	GroupChat   = 2;          	//群聊会话
	SystemChat  = 3;			//系统会话
};

message UserBasicInfo{
	uint32 user_id = 1;				// 用户id
	string user_name = 2;			// 用户名(用户自定义)
	uint32 user_gender = 3; 		// 用户性别,男：1 女：2  未知:0
	string user_nick_name = 4;		// 昵称
	string user_avatar_url = 5;		// 头像url
	string user_email = 6;			// 邮箱地址
	string user_cellphone = 7;		// 用户手机
    string sign_info = 8;			// 个性签名
};

message UserDetailInfo{
	uint32 user_id = 1;				// 用户id
	string user_name = 2;			// 用户名(用户自定义)
	uint32 user_gender = 3; 		// 用户性别,男：1 女：2  未知:0
	string user_nick_name = 4;		// 昵称
	string user_avatar_url = 5;		// 头像url
	string user_email = 6;			// 邮箱地址
	string user_cellphone = 7;		// 用户手机
    string sign_info = 8;			// 个性签名
	string birthday = 9; 			// 生日
	string school = 10;				// 学校
	string hometown = 11; 			// 家乡
	string career = 12; 			// 职业
	string describe = 13;			// 描述
};

message ServerIpAddr{
	string ip = 1;
	uint32 port = 2;
};
